cmd/controller/filter.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2018 OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"github.com/golang/glog"
+)
+
+// Filter contains name, state and filterInterface
+type Filter struct {
+	Name      string          // Name is the name of the filter
+	State     bool            // State is the State of the filter
+	Interface FilterInterface // Interface contains registerd filter
+}
+
+/*
+	ApplyFilter returns true if both any of include() or exclude() returns true.
+	We are having two types of filter function one is inclusion and exclusion type
+	if any of them returns true then filter doesn't want further process of that event.
+*/
+func (f *Filter) ApplyFilter(diskInfo *DiskInfo) bool {
+	return f.Interface.Include(diskInfo) || f.Interface.Exclude(diskInfo)
+
+}
+
+// Start implements FilterInterface's Start()
+func (f *Filter) Start() {
+	f.Interface.Start()
+}
+
+// FilterInterface contains ApplyFilter interface and Start()
+type FilterInterface interface {
+	Start()
+	ApplyFilter
```

Comment:
The struct ApplyFilter is clashing with the function ApplyFilter. Maybe the struct can be renamed to "Filters" or even totally eliminate it. bring up the Include and Exclude functions into the FilterInterace itself.

Reply:
done.